Yoast/wordpress-seo
# Add feature flag for Japanese support

## Context


* 

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Adds a feature-flag for Japanese support.

## Relevant technical choices:

* This PR is accompanied by another PR in Premium

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* In PHPStorm/your preferred IDE, navigate to `packages/yoastseo/src/languageProcessing/languages/_default` and copy the contents
* Inside the `languages` folder, create an `ja` folder and paste the contents from `_default` (don't commit any changes)
* Build the free plugin
* Enable the `JAPANESE_SUPPORT` feature by adding `define( 'YOAST_SEO_JAPANESE_SUPPORT', true );` in `basic-wordpress-config.php` file
* Set your site to Japanese` 日本語`
* Create/open a post
* In the console, enter `window.yoast.Researcher` and confirm that it returns as shown below (the Researcher file should start with `ja`):
<img width="537" alt="Screenshot 2021-09-09 at 12 15 22" src="https://user-images.githubusercontent.com/48715883/132668583-4066a162-49f7-415d-aafd-a2431865142b.png">

* Run this command `composer require yoast/wordpress-seo:dev-LINGO-1004-add-feature-flag-for-japanese@dev` in `wordpress-seo-premium`
* Build the premium plugin
* Reload your page
* In your console, confirm that the morphology request returns with `404`:
<img width="544" alt="Screenshot 2021-09-09 at 12 14 48" src="https://user-images.githubusercontent.com/48715883/132668643-2939e255-14d3-4232-b00a-ae04af88909e.png">


* Disable the `JAPANESE_SUPPORT` by adding this  `define( 'YOAST_SEO_JAPANESE_SUPPORT', false );` in `basic-wordpress-config.php` file
* Reload your page
* Confirm that you get the default Researcher by entering `window.yoast.Researcher` in the console:
<img width="538" alt="Screenshot 2021-09-09 at 12 17 44" src="https://user-images.githubusercontent.com/48715883/132668723-7e328f42-7333-46c0-98f0-09a435d4e5f5.png">


* Confirm that there is no morphology request
<img width="728" alt="Screenshot 2021-06-17 at 17 11 19" src="https://user-images.githubusercontent.com/48715883/122424791-49c74080-cf8f-11eb-8880-728b64259b8d.png">


### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

* Install and activate Free
* Enable the `JAPANESE_SUPPORT` feature by adding `define( 'YOAST_SEO_JAPANESE_SUPPORT', true );` in `basic-wordpress-config.php` file
* Set your site to Japanese` 日本語`
* Create/open a post
* In the console, enter `window.yoast.Researcher` and confirm that it returns as shown below (the Researcher file should start with `ja`):
<img width="537" alt="Screenshot 2021-09-09 at 12 15 22" src="https://user-images.githubusercontent.com/48715883/132668583-4066a162-49f7-415d-aafd-a2431865142b.png">

* Install and activate Premium
* Reload your page
* In your console, confirm that the morphology request returns with `404`:
<img width="544" alt="Screenshot 2021-09-09 at 12 14 48" src="https://user-images.githubusercontent.com/48715883/132668643-2939e255-14d3-4232-b00a-ae04af88909e.png">


* Disable the `JAPANESE_SUPPORT` by adding this  `define( 'YOAST_SEO_JAPANESE_SUPPORT', false );` in `basic-wordpress-config.php` file
* Reload your page
* Confirm that you get the default Researcher by entering `window.yoast.Researcher` in the console:
<img width="538" alt="Screenshot 2021-09-09 at 12 17 44" src="https://user-images.githubusercontent.com/48715883/132668723-7e328f42-7333-46c0-98f0-09a435d4e5f5.png">

* Confirm that there is no morphology request

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Documentation


## Quality assurance


Fixes https://yoast.atlassian.net/secure/RapidBoard.jspa?rapidView=99&modal=detail&selectedIssue=LINGO-1004
